sample_changer.rst

```diff
@@ -214,91 +139,109 @@ These are the functions that make up the sample changer API
     pass
 
 
-    def get_available_commands() -> List[List]:
-    """
-    Retrieves a List (of Lists)  of sample changer specific commands, The List
-    has the following structure:
-
-    ["Command-Category_1", [
-        ["cmd_1", "DisplayName", "Comment/Help-text"],
-        ...
-        ["cmd_n-1", "DisplayName", "Comment/Help-text"], 
-        ["cmd_n", "DisplayName", "Comment/Help-text"], 
-    ] 
-    ], 
-    ["Command-Category_n-1", [
-        ["cmd_1", "DisplayName", "Comment/Help-text"],
-        ...
-        ["cmd_n-1", "DisplayName", "Comment/Help-text"], 
-        ["cmd_n", "DisplayName", "Comment/Help-text"],
-    ]
-    ], 
-    ["Command-Category_n",  [
-        ...
-    ] 
-    ],  
-
-
-    :Example:
-    cmd_list = [
-    ["Actions",  [
-        ["home", "Home", "Actions"],
-        ["defreeze", "Defreeze gripper", "Actions"],
-        ["reset_sample_number", "Reset sample number", "Actions"],                   
-        ["change_gripper", "Change Gripper", "Actions"],
-        ["abort", "Abort", "Actions"],
-      ]
-    ]
-
-
-    :returns: List, of lists, on the format described above
-    :rtype List:
-    """
-    pass
-
-
-    def get_command_state() -> List[List]:
-    """
-    Returns a list with the currently available commands, which might depend on
-    sample changer state, and a status message:
-
-    [{"cmd_1": True if available else False,
-      "cmd_n": True if available else False,
-      "cmd_n-1": True if available else False},
-     message]
-
-     Where message is a str
-
-
-    :rtype: List
-    """
-    pass
-
-    
     def get_full_state() -> Dict:
     """
     :returns: A dictionary containing the complete state of the sample changer
 
     The returned dict has the following format:
 
-    {'state': GenericSampleChanger.SampleChangerState 
+    {'state': GenericSampleChanger.SampleChangerState
      'loaded_sample': LocationStr
      'contents': SampleNode
-     'cmds': "as returned by get_available_commands",
-     'commands_state': "as returned by get_command_state}",
+     'procedures': "as returned by get_procedures",
      'msg': "user message if any"
     }
 
     :rtype: dict
     """
 
+Specific procedures
+-------------------
 
-    def exec_command(cmd_name: str, *args, **kwargs) -> bool:
```

Comment:
Actually I think this function of the sample changer "name space" should sill exist, but use the same structures as defined in general. 

The idea would be to provide some kind of logic arrangement of the code, otherwise the risk is that everything would end up in general.

Reply:
That makes perfect sense.